### PR TITLE
FIx Git ratelimiting

### LINKED ISSUE #905: [feature] implement a github proxy to avoid rate limits
when downloading stuff from screenpipe's store i regularly get rate limited 

we could store a recent version of repo in an api and expose it

### PR DESCRIPTION
**Pull request by neo773**
This PR fixes rate limiting by using the `tree` GitHub API endpoint 
`https://api.github.com/repos/mediar-ai/screenpipe/git/trees/main?recursive=1`

This avoids recursive API calls and API response is cached via `http-cache-reqwest` it internally uses HTTP `e-tag` header present on API requests.

image

https://github.com/user-attachments/assets/42028ca1-e25d-4257-930f-e69f524088ef

/claim #905
/closes #905

### CODE REVIEW

**Comment by louis030195** [comment]
hmmmmmmmm

i thought u were going to do the git solution

so the biggest problems with pipes management now:
- rate limit 
- cannot preview the README without downloading 
- slow to download 
- anything else?



i see this PR helps the rate limit, does it improve the download speed?

**Comment by neo773** [comment]
I closed the old PR because git binary is not portable as it has other external dependencies so you can't embed it as a sidecar.

As for download speed improvement, yes this is much faster as it only makes a single API call to list all files and downloads them in parallel.

If there's a change in upstream repo the caching library will automatically invalidate old cache as the HTTP `e-tag` header would change.

**Comment by neo773** [comment]
Demo of downloading all the pipes in store

https://github.com/user-attachments/assets/49cb7042-d7d8-45a3-aeaa-83d6be96ac81

File: screenpipe-core/src/pipes.rs [hunk 1]
```diff
@@ -568,8 +570,19 @@ mod pipes {
         let dest_dir = dest_dir.to_path_buf();
 
         Box::pin(async move {
-            let client = Client::new();
+            // Create a cached client
+            let client = ClientBuilder::new(Client::new())
+                .with(Cache(HttpCache {
+                    mode: CacheMode::Default,
+                    manager: CACacheManager {
+                        path: home_dir().unwrap().join(".screenpipe/.http-cacache"),
```
Comment (louis030195) [inline screenpipe-core/src/pipes.rs:578]: this won't work on windows?

the /

use join

**Comment by louis030195** [comment]
/approve 

nice

**Comment by louis030195** [comment]
Uploading Screen Recording 2024-12-30 at 4.20.32 PM.mov…

okay i merged a bit fast - it does not work @neo773

**Comment by louis030195** [comment]
https://github.com/user-attachments/assets/113e91cd-a8af-4229-92ea-a5f2c9130ffa

**Comment by neo773** [comment]
Can you attach logs?

**Comment by louis030195** [comment]
Screenshot 2024-12-30 at 4 38 51 PM
Screenshot 2024-12-30 at 4 39 03 PM

**Comment by louis030195** [comment]
Screenshot 2024-12-30 at 4 40 35 PM

when i reload app it works

**Comment by neo773** [comment]
From your first screenshot seems like a Next.js issue with Turbopack failing to start?

Also I just tried couldn't reproduce it.

https://github.com/user-attachments/assets/3fed83eb-cbfc-40c5-9dda-ae18f34f81b9

**Comment by neo773** [comment]
Not able to reproduce it


https://github.com/user-attachments/assets/d41e5719-8a90-4f94-9375-1d425bd751fb


```logs
2024-12-31T01:03:52.080495Z  INFO screenpipe_audio::stt: device: Display 1 (output), speech ratio: 0, min_speech_ratio: 0.2, audio_frames: 512276, speech_frames: 0    
2024-12-31T01:03:54.425856Z  INFO screenpipe_audio::stt: device: MacBook Pro Microphone (input), speech ratio: 0, min_speech_ratio: 0.2, audio_frames: 512141, speech_frames: 0    
2024-12-31T01:03:56.308566Z  INFO screenpipe_server::resource_monitor: Runtime: 111s, Total Memory: 13% (4 GB / 33 GB), Total CPU: 168%, NPU: N/A
2024-12-31T01:04:01.498495Z  INFO screenpipe_core::pipes::pipes: [memories]  ○ Compiling / ...
2024-12-31T01:04:01.955201Z  INFO screenpipe_server::pipe_manager: stopping pipe: memories with pid 9702
2024-12-31T01:04:01.955258Z  INFO screenpipe_core::pipes::pipes: cleaning up 1 cron jobs for pipe memories
2024-12-31T01:04:01.955301Z  INFO screenpipe_core::pipes::pipes: stopped all cron jobs for pipe: memories
2024-12-31T01:04:01.955336Z  INFO screenpipe_core::pipes::pipes: shutting down cron job for pipe at path: /api/log
2024-12-31T01:04:02.057045Z  INFO screenpipe_server::pipe_manager: stopped pipe: memories
2024-12-31T01:04:02.727420Z ERROR screenpipe_core::pipes::pipes: [memories] FATAL: An unexpected Turbopack error occurred. Please report the content of /var/folders/_y/x4whdczj7mzf_2381zkvlmkc0000gn/T/next-panic-4a52adb856d8c81059dbfb3094a9a134.log, along with a description of what you were doing when the error occurred, to https://github.com/vercel/next.js/issues/new
2024-12-31T01:04:02.733917Z ERROR screenpipe_core::pipes::pipes: [memories] thread caused non-unwinding panic. aborting.
2024-12-31T01:04:04.852475Z ERROR screenpipe_server::server: failed to delete pipe: Directory not empty (os error 66)    
2024-12-31T01:04:04.852655Z ERROR tower_http::trace::on_failure: response failed classification=Status code: 500 Internal Server Error latency=2897 ms
2024-12-31T01:04:06.398781Z  INFO screenpipe_server::resource_monitor: Runtime: 121s, Total Memory: 7% (2 GB / 33 GB), Total CPU: 153%, NPU: N/A
2024-12-31T01:04:16.479311Z  INFO screenpipe_server::resource_monitor: Runtime: 131s, Total Memory: 7% (2 GB / 33 GB), Total CPU: 150%, NPU: N/A
2024-12-31T01:04:22.148763Z  INFO screenpipe_audio::stt: device: MacBook Pro Microphone (input), speech ratio: 0, min_speech_ratio: 0.2, audio_frames: 511956, speech_frames: 0    
2024-12-31T01:04:24.583267Z  INFO screenpipe_audio::stt: device: Display 1 (output), speech ratio: 0, min_speech_ratio: 0.2, audio_frames: 511956, speech_frames: 0    
2024-12-31T01:04:25.699265Z  INFO screenpipe_core::pipes::pipes: Processing pipe from source: https://github.com/mediar-ai/screenpipe/tree/main/pipes/memories
2024-12-31T01:04:26.561081Z  INFO screenpipe_server::resource_monitor: Runtime: 141s, Total Memory: 5% (2 GB / 33 GB), Total CPU: 160%, NPU: N/A
2024-12-31T01:04:26.741904Z  INFO screenpipe_core::pipes::pipes: pipe execution completed successfully [bun install]
2024-12-31T01:04:26.748984Z  INFO screenpipe_core::pipes::pipes: [bun install] bun install v1.1.42 (50eec002)
2024-12-31T01:04:30.739129Z  INFO screenpipe_core::pipes::pipes: [bun install] 
2024-12-31T01:04:30.739176Z  INFO screenpipe_core::pipes::pipes: [bun install] + @eslint/eslintrc@3.2.0
2024-12-31T01:04:30.739186Z  INFO screenpipe_core::pipes::pipes: [bun install] + @types/node@22.10.2
2024-12-31T01:04:30.739195Z  INFO screenpipe_core::pipes::pipes: [bun install] + @types/react@19.0.1
2024-12-31T01:04:30.739202Z  INFO screenpipe_core::pipes::pipes: [bun install] + @types/react-dom@19.0.2
2024-12-31T01:04:30.739231Z  INFO screenpipe_core::pipes::pipes: [bun install] + autoprefixer@10.4.20
2024-12-31T01:04:30.739244Z  INFO screenpipe_core::pipes::pipes: [bun install] + bun-types@1.1.42
2024-12-31T01:04:30.739287Z  INFO screenpipe_core::pipes::pipes: [bun install] + eslint@9.17.0
2024-12-31T01:04:30.739298Z  INFO screenpipe_core::pipes::pipes: [bun install] + eslint-config-next@15.1.0
2024-12-31T01:04:30.739305Z  INFO screenpipe_core::pipes::pipes: [bun install] + postcss@8.4.49
2024-12-31T01:04:30.739316Z  INFO screenpipe_core::pipes::pipes: [bun install] + tailwindcss@3.4.16
2024-12-31T01:04:30.739325Z  INFO screenpipe_core::pipes::pipes: [bun install] + typescript@5.7.2
2024-12-31T01:04:30.739337Z  INFO screenpipe_core::pipes::pipes: [bun install] + @radix-ui/react-accordion@1.2.2
2024-12-31T01:04:30.739348Z  INFO screenpipe_core::pipes::pipes: [bun install] + @radix-ui/react-checkbox@1.1.3
2024-12-31T01:04:30.739359Z  INFO screenpipe_core::pipes::pipes: [bun install] + @radix-ui/react-dialog@1.1.3
2024-12-31T01:04:30.739366Z  INFO screenpipe_core::pipes::pipes: [bun install] + @radix-ui/react-label@2.1.1
2024-12-31T01:04:30.739374Z  INFO screenpipe_core::pipes::pipes: [bun install] + @radix-ui/react-popover@1.1.3
2024-12-31T01:04:30.739381Z  INFO screenpipe_core::pipes::pipes: [bun install] + @radix-ui/react-progress@1.1.1
2024-12-31T01:04:30.739390Z  INFO screenpipe_core::pipes::pipes: [bun install] + @radix-ui/react-select@2.1.3
2024-12-31T01:04:30.739400Z  INFO screenpipe_core::pipes::pipes: [bun install] + @radix-ui/react-separator@1.1.1
2024-12-31T01:04:30.739413Z  INFO screenpipe_core::pipes::pipes: [bun install] + @radix-ui/react-slider@1.2.2
2024-12-31T01:04:30.739460Z  INFO screenpipe_core::pipes::pipes: [bun install] + @radix-ui/react-slot@1.1.1
2024-12-31T01:04:30.739474Z  INFO screenpipe_core::pipes::pipes: [bun install] + @radix-ui/react-switch@1.1.2
2024-12-31T01:04:30.739482Z  INFO screenpipe_core::pipes::pipes: [bun install] + @radix-ui/react-toast@1.2.3
2024-12-31T01:04:30.739493Z  INFO screenpipe_core::pipes::pipes: [bun install] + @radix-ui/react-tooltip@1.1.5
2024-12-31T01:04:30.739503Z  INFO screenpipe_core::pipes::pipes: [bun install] + @screenpipe/js@0.1.11
2024-12-31T01:04:30.739511Z  INFO screenpipe_core::pipes::pipes: [bun install] + @shadcn/ui@0.0.4
2024-12-31T01:04:30.739522Z  INFO screenpipe_core::pipes::pipes: [bun install] + @tanstack/react-query@5.62.7
2024-12-31T01:04:30.739532Z  INFO screenpipe_core::pipes::pipes: [bun install] + @types/js-levenshtein@1.1.3
2024-12-31T01:04:30.739543Z  INFO screenpipe_core::pipes::pipes: [bun install] + @types/lodash@4.17.13
2024-12-31T01:04:30.739552Z  INFO screenpipe_core::pipes::pipes: [bun install] + @types/react-syntax-highlighter@15.5.13
2024-12-31T01:04:30.739566Z  INFO screenpipe_core::pipes::pipes: [bun install] + ai@4.0.18
2024-12-31T01:04:30.739580Z  INFO screenpipe_core::pipes::pipes: [bun install] + class-variance-authority@0.7.1
2024-12-31T01:04:30.739591Z  INFO screenpipe_core::pipes::pipes: [bun install] + clsx@2.1.1
2024-12-31T01:04:30.739604Z  INFO screenpipe_core::pipes::pipes: [bun install] + cmdk@1.0.0
2024-12-31T01:04:30.739617Z  INFO screenpipe_core::pipes::pipes: [bun install] + date-fns@4.1.0
2024-12-31T01:04:30.739651Z  INFO screenpipe_core::pipes::pipes: [bun install] + framer-motion@11.14.4
2024-12-31T01:04:30.739671Z  INFO screenpipe_core::pipes::pipes: [bun install] + install@0.13.0
2024-12-31T01:04:30.739683Z  INFO screenpipe_core::pipes::pipes: [bun install] + js-levenshtein@1.1.6
2024-12-31T01:04:30.739695Z  INFO screenpipe_core::pipes::pipes: [bun install] + localforage@1.10.0
2024-12-31T01:04:30.739704Z  INFO screenpipe_core::pipes::pipes: [bun install] + lodash@4.17.21
2024-12-31T01:04:30.739715Z  INFO screenpipe_core::pipes::pipes: [bun install] + lucide-react@0.468.0
2024-12-31T01:04:30.739724Z  INFO screenpipe_core::pipes::pipes: [bun install] + magic-ui@0.1.0
2024-12-31T01:04:30.739740Z  INFO screenpipe_core::pipes::pipes: [bun install] + next@15.1.0
2024-12-31T01:04:30.739752Z  INFO screenpipe_core::pipes::pipes: [bun install] + npm@10.9.2
2024-12-31T01:04:30.739763Z  INFO screenpipe_core::pipes::pipes: [bun install] + ollama-ai-provider@1.1.0
2024-12-31T01:04:30.739775Z  INFO screenpipe_core::pipes::pipes: [bun install] + openai@4.76.3
2024-12-31T01:04:30.739789Z  INFO screenpipe_core::pipes::pipes: [bun install] + react@19.0.0
2024-12-31T01:04:30.739797Z  INFO screenpipe_core::pipes::pipes: [bun install] + react-day-picker@8.10.1
2024-12-31T01:04:30.739813Z  INFO screenpipe_core::pipes::pipes: [bun install] + react-dom@19.0.0
2024-12-31T01:04:30.739825Z  INFO screenpipe_core::pipes::pipes: [bun install] + react-markdown@9.0.1
2024-12-31T01:04:30.739839Z  INFO screenpipe_core::pipes::pipes: [bun install] + react-syntax-highlighter@15.6.1
2024-12-31T01:04:30.739847Z  INFO screenpipe_core::pipes::pipes: [bun install] + remark-gfm@4.0.0
2024-12-31T01:04:30.739859Z  INFO screenpipe_core::pipes::pipes: [bun install] + remark-math@6.0.0
2024-12-31T01:04:30.739871Z  INFO screenpipe_core::pipes::pipes: [bun install] + tailwind-merge@2.5.5
2024-12-31T01:04:30.739882Z  INFO screenpipe_core::pipes::pipes: [bun install] + tailwindcss-animate@1.0.7
2024-12-31T01:04:30.739894Z  INFO screenpipe_core::pipes::pipes: [bun install] 
2024-12-31T01:04:30.739906Z  INFO screenpipe_core::pipes::pipes: [bun install] 941 packages installed [3.99s]
2024-12-31T01:04:30.739921Z  INFO screenpipe_core::pipes::pipes: [bun install] 
2024-12-31T01:04:30.739934Z  INFO screenpipe_core::pipes::pipes: [bun install] Blocked 1 postinstall. Run `bun pm untrusted` for details.
2024-12-31T01:04:30.740170Z  INFO screenpipe_core::pipes::pipes: Detected Next.js project, setting up for production
2024-12-31T01:04:30.740546Z  INFO screenpipe_core::pipes::pipes: pipe copied successfully to: "/Users/neo/.screenpipe/pipes/memories"
2024-12-31T01:04:30.741303Z  INFO screenpipe_server::pipe_manager: pipe memories downloaded
2024-12-31T01:04:36.475132Z  INFO screenpipe_server::pipe_manager: pipe memories enabled
2024-12-31T01:04:36.476074Z  INFO screenpipe_core::pipes::pipes: installing dependencies for next.js pipe [memories]
2024-12-31T01:04:36.517910Z  INFO screenpipe_core::pipes::pipes: pipe execution completed successfully [memories]
2024-12-31T01:04:36.517983Z  INFO screenpipe_server::pipe_manager: started pipe: memories with pid 16698
2024-12-31T01:04:36.521580Z  INFO screenpipe_core::pipes::pipes: [memories] $ next dev --turbopack --port "18380"
2024-12-31T01:04:36.644324Z  INFO screenpipe_server::resource_monitor: Runtime: 151s, Total Memory: 3% (1 GB / 33 GB), Total CPU: 147%, NPU: N/A
2024-12-31T01:04:38.692180Z  INFO screenpipe_core::pipes::pipes: [memories]    ▲ Next.js 15.1.0 (Turbopack)
2024-12-31T01:04:38.692230Z  INFO screenpipe_core::pipes::pipes: [memories]    - Local:        http://localhost:18380
2024-12-31T01:04:38.692241Z  INFO screenpipe_core::pipes::pipes: [memories]    - Network:      http://192.168.0.148:18380
2024-12-31T01:04:38.692250Z  INFO screenpipe_core::pipes::pipes: [memories] 
2024-12-31T01:04:38.692286Z  INFO screenpipe_core::pipes::pipes: [memories]  ✓ Starting...
2024-12-31T01:04:39.241461Z  INFO screenpipe_core::pipes::pipes: [memories]  ✓ Ready in 700ms
2024-12-31T01:04:39.243302Z  INFO screenpipe_core::pipes::pipes: [memories]  ⚠ Webpack is configured while Turbopack is not, which may cause problems.
2024-12-31T01:04:39.243334Z  INFO screenpipe_core::pipes::pipes: [memories]  ⚠ See instructions if you need to configure Turbopack:
2024-12-31T01:04:39.243348Z  INFO screenpipe_core::pipes::pipes: [memories]   https://nextjs.org/docs/app/api-reference/next-config-js/turbo
2024-12-31T01:04:39.243359Z  INFO screenpipe_core::pipes::pipes: [memories] 
2024-12-31T01:04:40.274157Z  INFO screenpipe_core::pipes::pipes: [memories]  ○ Compiling / ...
2024-12-31T01:04:42.090176Z  INFO screenpipe_core::pipes::pipes: [memories]  ✓ Compiled / in 2.3s
2024-12-31T01:04:42.247913Z  INFO screenpipe_core::pipes::pipes: [memories]  GET / 200 in 2485ms
2024-12-31T01:04:42.606406Z  INFO screenpipe_server::server: received search request: query='', content_type=OCR, limit=2, offset=0, start_time=Some(2024-12-31T00:01:28.526Z), end_time=Some(2024-12-31T02:01:28.526Z), app_name=None, window_name=None, min_length=None, max_length=None, speaker_ids=None    
2024-12-31T01:04:42.894952Z  INFO screenpipe_core::pipes::pipes: [memories]  ✓ Compiled /favicon.ico in 236ms
2024-12-31T01:04:42.937185Z  INFO screenpipe_core::pipes::pipes: [memories]  GET /favicon.ico?favicon.6178839d.ico 200 in 281ms
2024-12-31T01:04:44.849119Z  WARN sqlx::query: summary="SELECT COUNT(DISTINCT frames.id) FROM …" db.statement="\n\nSELECT\n  COUNT(DISTINCT frames.id)\nFROM\n  ocr_text_fts\n  JOIN ocr_text ON ocr_text_fts.frame_id = ocr_text.frame_id\n  JOIN frames ON ocr_text.frame_id = frames.id\nWHERE\n  1 = 1\n  AND (\n    ?2 IS NULL\n    OR frames.timestamp >= ?2\n  )\n  AND (\n    ?3 IS NULL\n    OR frames.timestamp <= ?3\n  )\n  AND (\n    ?4 IS NULL\n    OR ocr_text.app_name LIKE '%' || ?4 || '%'\n  )\n  AND (\n    ?5 IS NULL\n    OR ocr_text.window_name LIKE '%' || ?5 || '%'\n  )\n  AND (\n    ?6 IS NULL\n    OR LENGTH(ocr_text.text) >= ?6\n  )\n  AND (\n    ?7 IS NULL\n    OR LENGTH(ocr_text.text) <= ?7\n  )\n" rows_affected=1 rows_returned=1 elapsed=2.236709708s
2024-12-31T01:04:45.323587Z  WARN sqlx::query: summary="SELECT ocr_text.frame_id, ocr_text.text as …" db.statement="\n\nSELECT\n  ocr_text.frame_id,\n  ocr_text.text as ocr_text,\n  ocr_text.text_json,\n  frames.timestamp,\n  video_chunks.file_path,\n  frames.offset_index,\n  ocr_text.app_name,\n  ocr_text.ocr_engine,\n  ocr_text.window_name,\n  GROUP_CONCAT(tags.name, ',') as tags\nFROM\n  ocr_text\n  JOIN frames ON ocr_text.frame_id = frames.id\n  JOIN video_chunks ON frames.video_chunk_id = video_chunks.id\n  LEFT JOIN vision_tags ON frames.id = vision_tags.vision_id\n  LEFT JOIN tags ON vision_tags.tag_id = tags.id\nWHERE\n  1 = 1\n  AND ocr_text.text != 'No text found'\n  AND (\n    ?2 IS NULL\n    OR frames.timestamp >= ?2\n  )\n  AND (\n    ?3 IS NULL\n    OR frames.timestamp <= ?3\n  )\n  AND (\n    ?4 IS NULL\n    OR LENGTH(ocr_text.text) >= ?4\n  )\n  AND (\n    ?5 IS NULL\n    OR LENGTH(ocr_text.text) <= ?5\n  )\n  AND (\n    ?6 IS NULL\n    OR ocr_text.app_name LIKE '%' || ?6 || '%' COLLATE NOCASE\n  )\n  AND (\n    ?7 IS NULL\n    OR ocr_text.window_name LIKE '%' || ?7 || '%' COLLATE NOCASE\n  )\nGROUP BY\n  ocr_text.frame_id\nORDER BY\n  frames.timestamp DESC\nLIMIT\n  ?8 OFFSET ?9\n" rows_affected=1 rows_returned=2 elapsed=2.708128292s
2024-12-31T01:04:45.323878Z  INFO screenpipe_server::server: search completed: found 608 results    
2024-12-31T01:04:45.326319Z  INFO screenpipe_server::server: received search request: query='', content_type=OCR, limit=2, offset=0, start_time=Some(2024-12-17T17:39:34.861Z), end_time=Some(2024-12-17T19:39:34.861Z), app_name=None, window_name=None, min_length=None, max_length=None, speaker_ids=None    
2024-12-31T01:04:45.683383Z  INFO screenpipe_server::server: search completed: found 5503 results    
2024-12-31T01:04:45.686009Z  INFO screenpipe_server::server: received search request: query='', content_type=OCR, limit=2, offset=0, start_time=Some(2024-12-29T03:46:57.098Z), end_time=Some(2024-12-29T05:46:57.098Z), app_name=None, window_name=None, min_length=None, max_length=None, speaker_ids=None    
2024-12-31T01:04:45.877625Z  INFO screenpipe_server::server: search completed: found 0 results    
2024-12-31T01:04:45.880001Z  INFO screenpipe_server::server: received search request: query='', content_type=OCR, limit=2, offset=0, start_time=Some(2024-12-10T07:38:12.858Z), end_time=Some(2024-12-10T09:38:12.858Z), app_name=None, window_name=None, min_length=None, max_length=None, speaker_ids=None    
2024-12-31T01:04:46.148916Z  INFO screenpipe_server::server: search completed: found 0 results    
2024-12-31T01:04:46.151220Z  INFO screenpipe_server::server: received search request: query='', content_type=OCR, limit=2, offset=0, start_time=Some(2024-12-28T11:28:29.471Z), end_time=Some(2024-12-28T13:28:29.471Z), app_name=None, window_name=None, min_length=None, max_length=None, speaker_ids=None    
2024-12-31T01:04:46.364285Z  INFO screenpipe_server::server: search completed: found 1283 results    
2024-12-31T01:04:46.742377Z  INFO screenpipe_server::resource_monitor: Runtime: 161s, Total Memory: 3% (1 GB / 33 GB), Total CPU: 147%, NPU: N/A
2024-12-31T01:04:46.843720Z  INFO screenpipe_core::pipes::pipes: [memories]  POST / 200 in 258ms
2024-12-31T01:04:47.037399Z  INFO screenpipe_core::pipes::pipes: [memories]  POST / 200 in 308ms
2024-12-31T01:04:47.127231Z  INFO screenpipe_core::pipes::pipes: [memories]  POST / 200 in 445ms
2024-12-31T01:04:47.183599Z  INFO screenpipe_core::pipes::pipes: [memories]  POST / 200 in 272ms
2024-12-31T01:04:47.303009Z  INFO screenpipe_core::pipes::pipes: [memories]  POST / 200 in 200ms
2024-12-31T01:04:47.377324Z  INFO screenpipe_core::pipes::pipes: [memories]  POST / 200 in 406ms
2024-12-31T01:04:47.568847Z  INFO screenpipe_core::pipes::pipes: [memories]  POST / 200 in 279ms
2024-12-31T01:04:47.728481Z  INFO screenpipe_core::pipes::pipes: [memories]  POST / 200 in 200ms
```

**Comment by louis030195** [comment]
i did not add the url in the input in my video

**Comment by louis030195** [comment]
something broke when pipe have different id than dir maybe

**Comment by louis030195** [comment]
fixed like this https://github.com/mediar-ai/screenpipe/commit/1105ac6f2de7652a1f15afca33b25491049bd776

but shouldnt happen - should work at rust side i guess?

**Comment by neo773** [comment]
This is odd still can't reproduce

https://github.com/user-attachments/assets/bcd9ead2-5158-49e1-a686-9ba24a38d8e5

**Pull request merged**